### PR TITLE
Api building

### PR DESCRIPTION
**Pull request by perryr16**
## Sinatra
* Set a few routes
* Build out the API response

### CODE REVIEW

File: i-scream-arcade-microservice-app.rb [hunk 1]
```diff
@@ -1 +1,38 @@
...
+# before do
+#  content_type :json
+# end
+
+# game_title = "Mario$20Kart%2064"
+results = GameResults.new
+
+get '/game/:game_title' do
+  json results.game_params_by_name(params[:game_title])
+end
```
Comment (perryr16) [inline i-scream-arcade-microservice-app.rb:20]: These are similar to rails routes, the last line is returned in the view

File: poros/game_results.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
+
+class GameResults
+  
+  def service 
+    GameService.new
+  end
+
+  def game_params_by_name(game_name)
+    params = service.get_games_by_name(game_name)[0]
```
Comment (perryr16) [inline poros/game_results.rb:9]: get_games_by_name() will return the api response for `api.../games` `where name = game_name`. There are more values returned than needed

File: poros/game_results.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    binding.pry
+    result = {data: {
+      age_ratings:  run_array(:get_age_ratings, :rating, params[:age_ratings]), # array of ids
+      release_date: service.get_release_date(params[:first_release_date]), # id
+      cover:        cover_url(params[:cover]),
+      popularity:   params[:popularity], # good
+      summary:      params[:summary], # good
```
Comment (perryr16) [inline poros/game_results.rb:15]: some values are returned as usable data like summary as a string and popularity as a float. but others are just lists of ids [1,10, 43...]

File: poros/game_results.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      name:         params[:name], # good
+      total_rating: params[:total_rating], # good
+      categories:   params[:category], # got 0
+      genres:       run_array(:get_game_genres, :name, params[:genres]), # array of ids
+      keywords:     params[:keywords], # array of ids -- several hundred :(
+      platforms:    run_array(:get_platforms, :name, params[:platforms]), # array of ids
+      similars:     run_array(:get_games_by_id, :name, params[:similar_games]), # array of ids
+      screenshots:  run_array(:get_screenshots, :url, params[:screenshots]), # array of ids
+      themes:       run_array(:get_game_themes, :name, params[:themes]), # array of ids
+      videos:       youtube_id(params[:videos][0]) # good -> array of ids
```
Comment (perryr16) [inline poros/game_results.rb:25]: Things like youtube_id only return one value, and we can hit the API (again) to return the video_id if we pass in the `video_id` (given by the api)

File: poros/game_results.rb [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      release_date: service.get_release_date(params[:first_release_date]), # id
+      cover:        cover_url(params[:cover]),
+      popularity:   params[:popularity], # good
+      summary:      params[:summary], # good
+      name:         params[:name], # good
+      total_rating: params[:total_rating], # good
+      categories:   params[:category], # got 0
+      genres:       run_array(:get_game_genres, :name, params[:genres]), # array of ids
+      keywords:     params[:keywords], # array of ids -- several hundred :(
+      platforms:    run_array(:get_platforms, :name, params[:platforms]), # array of ids
```
Comment (perryr16) [inline poros/game_results.rb:21]: other return values are full arrays of ids `[1,10,43...]`. We can then hit the API again to return the details of a single id. If there are many ids, then I use the method `run_array` (general, I know) to iterate through that array of ids and return the name of each

File: poros/game_results.rb [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  
+  def youtube_id(video_id)
+    service.get_game_videos(video_id)[0][:video_id]
+  end
+
+  def keyword_id(word)
+    {data: service.get_keyid(word)[0]}
+  end
+
+  def generate_ids(words)
```
Comment (perryr16) [inline poros/game_results.rb:47]: not used

File: services/game_service.rb [hunk 7]
```diff
@@ -0,0 +1,108 @@
+require 'faraday'
+
+class GameService
+
+GAME_API = '177f68493b7f48e7c3deaf6d5a815ff0'
```
Comment (perryr16) [inline services/game_service.rb:5]: Wasnt sure how to use environment variables in Sinatra

File: services/game_service.rb [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+  end
+
+  def get_keyid(word)
+    response = conn.get("/keywords") do |res|
+      res.body = "fields id, name; where name = \"#{word}\";" 
+    end
+    body = JSON.parse(response.body, symbolize_names: true)
+  end
+
+  def get_game_videos(video_id)
```
Comment (perryr16) [inline services/game_service.rb:40]: these methods are all VERY similar, could be refactored to DRY it up

File: poros/game_results.rb (hunk 4 above)
Comment (melatran) [reply poros/game_results.rb:25]: this is very helpful to know

**Review by perryr16 — COMMENTED** [review]
I am open to any and all refactors

File: services/game_service.rb (hunk 8 above)
Comment (melatran) [reply services/game_service.rb:40]: yeah, i was wondering if we could do like a get_url method and then pass in the url path we want

**Comment by melatran** [comment]
This looks like a great start to finally being able to interact with the our API. Thanks for setting it up

**Review by melatran — APPROVED** [review]

**Pull request merged**